### PR TITLE
Rdoc-1897 data subscriptions stats

### PR DESCRIPTION
**Pull request by reebhub**
new Studio stats page
(to allow the documentation of the data subscriptions stats view)

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/stats-view.markdown [hunk 1]
```diff
@@ -0,0 +1,64 @@
+﻿# Stats View
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/stats-view.markdown:1]: => No need for this dedicated page - Remove

The menu should be:

Stats > Stats
Stats > IO Stats
Stats > Storage Report
Stats > Running Queries
Stats > Ongoing Tasks Stats

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 2]
```diff
@@ -0,0 +1,160 @@
+﻿# Ongoing Tasks Stats View
+---
+
+{NOTE: }
+
+* The **Ongoing Tasks Stats** view monitors ongoing _Replication_, _ETL_, and _Data Subscription_ tasks.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:6]: Replication => External Replication
ETL => RavenDB ETL

Replace with:
=>
The Ongoing Tasks Stats view monitors the following ongoing tasks:
* External Replication
* RavenDB ETL
* Subscriptions

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 3]
```diff
@@ -0,0 +1,160 @@
+﻿# Ongoing Tasks Stats View
+---
+
+{NOTE: }
+
+* The **Ongoing Tasks Stats** view monitors ongoing _Replication_, _ETL_, and _Data Subscription_ tasks.  
+* The monitored tasks' overall performance is shown as a graph in a continuous timeline ruler.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:7]: Replace:
=>
The tasks activities are graphically displayed for selected time frames

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 4]
```diff
@@ -0,0 +1,160 @@
+﻿# Ongoing Tasks Stats View
+---
+
+{NOTE: }
+
+* The **Ongoing Tasks Stats** view monitors ongoing _Replication_, _ETL_, and _Data Subscription_ tasks.  
+* The monitored tasks' overall performance is shown as a graph in a continuous timeline ruler.  
+* The timeline selection tool allows you to choose any timeline segment and explore task activities
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:8]: Remove this bullet
It is explained later in the relevant section, see my comments below

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+{NOTE: }
+
+* The **Ongoing Tasks Stats** view monitors ongoing _Replication_, _ETL_, and _Data Subscription_ tasks.  
+* The monitored tasks' overall performance is shown as a graph in a continuous timeline ruler.  
+* The timeline selection tool allows you to choose any timeline segment and explore task activities 
+  during this period.  
+* Activities related to each monitored task are displayed over a ruler dedicated to this task.  
+  Task rulers show task activities during the time selected by the timeline select tool.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:11]: Remove this bullet

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  Task rulers show task activities during the time selected by the timeline select tool.  
+
+* In this page:  
+  * [Ongoing Tasks Stats View](../../../studio/database/stats/ongoing-tasks-stats#ongoing-tasks-stats-view)  
+  * [The Timeline Ruler](../../../studio/database/stats/ongoing-tasks-stats#the-timeline-ruler)  
+  * [Specific Tasks View](../../../studio/database/stats/ongoing-tasks-stats#specific-tasks-view)  
+     * [Task Rulers](../../../studio/database/stats/ongoing-tasks-stats#task-rulers)  
+     * [Basic Task View](../../../studio/database/stats/ongoing-tasks-stats#basic-task-view)  
+     * [Expanded Task View](../../../studio/database/stats/ongoing-tasks-stats#expanded-task-view)  
+     * [Additional Information](../../../studio/database/stats/ongoing-tasks-stats#additional-information)
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:20]: Menu should be:

- Ongoing Tasks Stats View
- The Timeline Ruler
- Task Rulers

Explained more in the comments inside those sections

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:6]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:7]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:8]: removed

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:11]: removed

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown (hunk 6 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:20]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+{PANEL: Ongoing Tasks Stats View}
+
+![Ongoing Tasks Stats View](images/ongoing-tasks-stats-view.png "Ongoing Tasks Stats View")
+
+* **1.** Ongoing Tasks Stats  
+  Click to open the Ongoing Tasks Stats View.  
+
+* **2.** Database  
+  Click to choose the database whose ongoing tasks you want to monitor.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:36]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+* **2.** Database  
+  Click to choose the database whose ongoing tasks you want to monitor.  
+
+{PANEL/}
+
+{PANEL: The Timeline Ruler}
+
+![The Timeline Ruler](images/the-timeline-ruler.png "The Timeline Ruler")
+
+* The timeline ruler will appear after data is collected from ongoing tasks.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:44]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+![The Timeline Ruler](images/the-timeline-ruler.png "The Timeline Ruler")
+
+* The timeline ruler will appear after data is collected from ongoing tasks.  
+
+---
+
+![Clear Graph and Selection](images/selection-and-clear-buttons.png "Clear Graph and Selection")
+
+* **1.** Selection Frame  
+  Click and Drag to select a segment of the timeline.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:51]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+* The timeline ruler will appear after data is collected from ongoing tasks.  
+
+---
+
+![Clear Graph and Selection](images/selection-and-clear-buttons.png "Clear Graph and Selection")
+
+* **1.** Selection Frame  
+  Click and Drag to select a segment of the timeline.  
+* **2.** Clear Selection  
+  Click to clear your current selection frame.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:53]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+![Clear Graph and Selection](images/selection-and-clear-buttons.png "Clear Graph and Selection")
+
+* **1.** Selection Frame  
+  Click and Drag to select a segment of the timeline.  
+* **2.** Clear Selection  
+  Click to clear your current selection frame.  
+* **3.** Clear Graph  
+  Click to clear the collected data.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:55]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+  Click to clear your current selection frame.  
+* **3.** Clear Graph  
+  Click to clear the collected data.  
+
+---
+
+![Tail Stream End](images/monitor-tail.png "Tail Stream End")
+
+* **Monitoring (tail -f)**  
+  When this option is checked Studio continuously relocates the selection frame at the end of the data stream.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:62]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+{PANEL/}
+
+{PANEL: Specific Tasks View}
+
+---
+
+### Task Rulers
+
+A ruler dedicated to each monitored task displays the task's activities during the 
+time specified by the timeline selection box.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:73]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+### Task Rulers
+
+A ruler dedicated to each monitored task displays the task's activities during the 
+time specified by the timeline selection box.  
+
+![Task Rulers](images/task-rulers.png "Task Rulers")  
+
+* **1.** ETL Task Ruler
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:77]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+   * Batches size  
+* **Documents Batch**  
+  Details related to the monitored batch:  
+   * Batch duration  
+   * Number of documents sent in this batch  
+   * Transferred documents size  
+   * Number of [included](../../../client-api/data-subscriptions/creation/examples#create-subscription-with-include-statement) 
+     documents, counters, and time series  
+* **Sending Documents**  
+  Documents delivery duration
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:117]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+  Details related to the monitored batch:  
+   * Batch duration  
+   * Number of documents sent in this batch  
+   * Transferred documents size  
+   * Number of [included](../../../client-api/data-subscriptions/creation/examples#create-subscription-with-include-statement) 
+     documents, counters, and time series  
+* **Sending Documents**  
+  Documents delivery duration  
+* **Waiting for ACK**  
+  How long did it take for the delivery to be acknowledged?
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:119]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+
+### Additional Information
+
+Additional information that can be learned from task views includes:  
+
+* **Active** and **Inactive** tasks are represented in different styles.  
+   * An active task:  
+    ![Active Task](images/active-task.png "Active Task")  
+   * An inactive task:  
+    ![Inactive Task](images/inactive-task.png "Inactive Task")
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:131]: moved to the Task Rulers panel

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/stats-view.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/stats-view.markdown:1]: - first panel removed
- second panel updated

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+
+### Task Rulers
+
+A ruler dedicated to each monitored task displays the task's activities during the 
+time specified by the timeline selection box.  
+
+![Task Rulers](images/task-rulers.png "Task Rulers")  
+
+* **1.** ETL Task Ruler  
+* **2,3,4.** Data Subscription Tasks Rulers
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:78]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+time specified by the timeline selection box.  
+
+![Task Rulers](images/task-rulers.png "Task Rulers")  
+
+* **1.** ETL Task Ruler  
+* **2,3,4.** Data Subscription Tasks Rulers  
+
+---
+
+### Basic Task View
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:82]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+![Basic Task View](images/basic-task-view.png "Basic Task View")
+
+* **1.** Chosen time segment  
+* **2.** ETL task basic view  
+    ![ETL Basic View](images/etl-basic-view.png "ETL Basic View")  
+* **3.** Data Subscription task basic view  
+
+---
+
+### Expanded Task View
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:95]: done

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+  hovering over the mark provides an explanation for this event.  
+   * A forcibly-aborted connection:  
+    ![Forcibly-Aborted Connnection](images/forcibly-aborted-connection.png "Forcibly-Aborted Connnection")  
+   * A rejected connection:  
+    ![Rejected Connnection](images/rejected-connection.png "Rejected Connnection")  
+      * **Rejected Connection**  
+        In this case, a client was rejected for trying to connect a subscription that 
+        was already in use by another client.  
+        Hovering over the rejection mark pops up a message explaining that the subscription 
+        was occupied.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/stats/ongoing-tasks-stats.markdown:143]: done

**Pull request closed without merging**